// @chainlink/contracts/ethers/v0.4/ChainlinkedFactory.ts
/* Generated by ts-generator ver. 0.0.8 */
/* tslint:disable */

import { Contract, ContractFactory, Signer } from "ethers";
import { Provider } from "ethers/providers";
import { UnsignedTransaction } from "ethers/utils/transaction";

import { TransactionOverrides } from ".";
import { Chainlinked } from "./Chainlinked";

export class ChainlinkedFactory extends ContractFactory {
  constructor(signer?: Signer) {
    super(_abi, _bytecode, signer);
  }

  deploy(overrides?: TransactionOverrides): Promise<Chainlinked> {
    return super.deploy(overrides) as Promise<Chainlinked>;
  }
  getDeployTransaction(overrides?: TransactionOverrides): UnsignedTransaction {
    return super.getDeployTransaction(overrides);
  }
  attach(address: string): Chainlinked {
    return super.attach(address) as Chainlinked;
  }
  connect(signer: Signer): ChainlinkedFactory {
    return super.connect(signer) as ChainlinkedFactory;
  }
  static connect(
    address: string,
    signerOrProvider: Signer | Provider
  ): Chainlinked {
    return new Contract(address, _abi, signerOrProvider) as Chainlinked;
  }
}

const _abi = [
  {
    anonymous: false,
    inputs: [
      {
        indexed: true,
        name: "id",
        type: "bytes32"
      }
    ],
    name: "ChainlinkRequested",
    type: "event"
  },
  {
    anonymous: false,
    inputs: [
      {
        indexed: true,
        name: "id",
        type: "bytes32"
      }
    ],
    name: "ChainlinkFulfilled",
    type: "event"
  },
  {
    anonymous: false,
    inputs: [
      {
        indexed: true,
        name: "id",
        type: "bytes32"
      }
    ],
    name: "ChainlinkCancelled",
    type: "event"
  }
];

const _bytecode =
  "0x60806040526001600455348015601457600080fd5b5060358060226000396000f3006080604052600080fd00a165627a7a72305820ec447023871083602e1c992f5493b94e11a84ec300d6d25190228a1cca03f5eb0029";
